### PR TITLE
add support for PHP8

### PR DESCRIPTION
**Pull request by ecourtial**
#54 

- Add support for PHP8.0
- Remove blocking issue added in https://github.com/wizaplace/php-etl/pull/4/files (so, we might have a BC if someone rely on this one)

May be it is time to consider a 2.0 to drop all the old stuff and have a fresh base for new features.

### CODE REVIEW

File: src/Row.php [hunk 1]
```diff
@@ -151,11 +151,9 @@ public function __get(string $key)
      *
```
Comment (ecourtial) [inline src/Row.php:151]: Technically this is a BC... because we introduced this issue earlier.

**Review by kdebisschop — APPROVED** [review]
We probably cannot say we're fully ready for PHP 8 when we expect authors to use php-cs-fixer quality profiles and php-cs-fixer itself will not run on php 8. But this is nice.

The one comment I made is truly optional.

**Comment by kdebisschop** [comment]
In response to your comment above -- I'd support a 2.0 release that used `"php": "~7.4|~8.0"` as a basis. With the availability of typed properties in 7.4, we can further reduce the need for PHPdoc annotations for types, but doing so is interface changing, so increasing to 2.x is helpful.

It might be nice to tag a release where we are before making that jump.

**Comment by ecourtial** [comment]
@kdebisschop ok !
So to sum-up:

- we merge this PR and tag the 1.3 version with the mention "partial support for PHP8 - partial meaning for the dev tools", and some details.
- We can prepare a 2.0 branch to start working on it with a minimal support of 7.4 (or 8 directly ?).

File: tests/Extractors/QueryTest.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ public function custom_connection_and_bindings()
         $extractor->input('select query');
         $extractor->options([
             'connection' => 'connection',
-            'bindings' => 'bindings',
+            'bindings' => ['bindings'],
```
[the remark this answers is not in the document]
Comment (ecourtial) [reply tests/Extractors/QueryTest.php:60]: Yes !

[the remark this answers is not in the document]
Comment (ecourtial) [reply tests/Extractors/QueryTest.php:60]: In fact now. Would need to do a BC in the Extractor itself.

**Pull request merged**